nutridine/src/screens/macro/MacroPage.js

```diff
@@ -105,133 +106,153 @@ function AttributeSliders() {
-                  <Text fontFamily={"navbar"} fontWeight={"500"}>
-                    Min
-                  </Text>
-                  <NumberInput
-                    value={macros[name][0]}
-                    min={min}
-                    max={macros[name][1]}
-                    onChange={(valueString) =>
-                      handleSliderChange(name, [
-                        Number(valueString),
-                        macros[name][1],
-                      ])
-                    }
-                    keepWithinRange={true}
-                    clampValueOnBlur={false}
-                    width={"100px"}
+      {
+        showSliders ?
+          <VStack spacing={5} p={0} align="stretch" px={[8, 8, 0]}>
+            <Box>
+              <Text fontSize={"xl"} mb={"3"}>
+                Macronutrient Filter
+              </Text>
+              <Text
+                fontFamily={"navbar"}
+                fontSize={["md", "md", "lg", "lg"]}
+              >
+                Set your desired macronutrient ranges and click 'Submit' to search for matching foods.
+              </Text>
+            </Box>
+            {macroSettings.map(({ name, label, min, max }) => (
+              <Card
+                key={name}
+                width="100%"
+                borderWidth="1px"
+                borderColor={borderColor}
+                borderRadius="15"
+                overflow="hidden"
+                bg={cardBg}
+                fontFamily={"navbar"}
+              >
+                <CardBody>
+                  <Text
+                    fontSize={["md", "md", "lg", "xl"]}
+                    fontFamily={"navbar"}
+                    mb={2}
                   >
-                    <NumberInputField />
-                    <NumberInputStepper>
-                      <NumberIncrementStepper />
-                      <NumberDecrementStepper />
-                    </NumberInputStepper>
-                  </NumberInput>
-                  <Text fontFamily={"navbar"} fontWeight={"500"}>
-                    Max
+                    {label}
                   </Text>
-                  <NumberInput
-                    value={macros[name][1]}
-                    min={macros[name][0]}
+                  <RangeSlider
+                    aria-label={[`min-${name}`, `max-${name}`]}
+                    onChange={(val) => handleSliderChange(name, val)}
+                    value={macros[name]}
+                    min={min}
                     max={max}
-                    onChange={(valueString) =>
-                      handleSliderChange(name, [
-                        macros[name][0],
-                        Number(valueString),
-                      ])
-                    }
-                    keepWithinRange={true}
-                    clampValueOnBlur={false}
-                    width={"100px"}
+                    colorScheme={"green"}
+                    minStepsBetweenThumbs={20}
```

Comment:
This will distance the thumbs by 20, then the user wonr be able to set a smaller range for things like protein for example. maybe they want a range between 20-25